activemodel/test/cases/validations/comparison_validation_test.rb

```diff
@@ -0,0 +1,242 @@
+    Topic.validates_comparison_of :approved, other_than: date_value
+
+    invalid!([Date.parse("2020-08-02"), DateTime.new(2020, 8, 2, 0, 0)], "must be other than 2020-08-02")
+    valid!([
+      Date.parse("2019-08-03"),
+      Date.parse("2020-07-03"),
+      Date.parse("2020-08-01"),
+      DateTime.new(2020, 8, 1, 12, 34),
+      Date.parse("2020-08-03"),
+      DateTime.new(2020, 8, 2, 12, 34)])
+  end
+
+  def test_validates_comparison_with_proc
+    Topic.define_method(:requested) { Date.new(2020, 8, 1) }
+    Topic.validates_comparison_of :approved, greater_than_or_equal_to: Proc.new(&:requested)
+
+    invalid!([Date.new(2020, 7, 1), Date.new(2019, 7, 1), DateTime.new(2020, 7, 1, 22, 34)])
+    valid!([Date.new(2020, 8, 2), DateTime.new(2021, 8, 1)])
+  ensure
+    Topic.remove_method :requested
+  end
+
+  def test_validates_comparison_with_method
+    Topic.define_method(:requested) { Date.new(2020, 8, 1) }
+    Topic.validates_comparison_of :approved, greater_than_or_equal_to: :requested
+
+    invalid!([Date.new(2020, 7, 1), Date.new(2019, 7, 1), DateTime.new(2020, 7, 1, 22, 34)])
+    valid!([Date.new(2020, 8, 2), DateTime.new(2021, 8, 1)])
+  ensure
+    Topic.remove_method :requested
+  end
+
+  def test_validates_comparison_with_custom_compare
+    custom = Struct.new(:amount) {
+      include Comparable
+
+      def <=>(other)
+        amount % 100 <=> other.amount % 100
+      end
+    }
+    Topic.validates_comparison_of :approved, greater_than_or_equal_to: custom.new(1150)
+
+    invalid!([custom.new(530), custom.new(2325)])
+    valid!([custom.new(575), custom.new(250), custom.new(1999)])
+  end
+
+  def test_validates_comparison_with_blank_allowed
+    Topic.validates_comparison_of :approved, greater_than: "cat", allow_blank: true
+
+    invalid!(["ant"])
+    valid!([nil, ""])
+  end
+
+  def test_validates_comparison_with_nil_allowed
+    Topic.validates_comparison_of :approved, less_than: 100, allow_nil: true
+
+    invalid!([200])
+    valid!([nil, 50])
+  end
+
+  def test_validates_comparison_of_incomparables
+    Topic.validates_comparison_of :approved, less_than: "cat"
+
+    invalid!([12], "comparison of Integer with String failed")
+    invalid!([nil])
+    valid!([])
+  end
+
+  def test_validates_comparison_of_multiple_values
+    Topic.validates_comparison_of :approved, other_than: 17, greater_than: 13
+
+    invalid!([12, nil, 17])
+    valid!([15])
+  end
+
+  def test_validates_comparison_of_no_options
+    Topic.validates_comparison_of :approved
+
+    invalid!([])
+    valid!([15, 12, nil, 17])
+  end
+
+  private
+    def invalid!(values, error = nil)
+      with_each_topic_approved_value(values) do |topic, value|
+        assert topic.invalid?, "#{value.inspect} failed comparison"
+        assert topic.errors[:approved].any?, "FAILED for #{value.inspect}"
+        assert_equal error, topic.errors[:approved].first if error
+      end
+    end
+
+    def valid!(values)
```

Comment:
```suggestion
    def assert_valid_values(values)
```

Reply:
This was intended to match NumericalityValidator's tests:
https://github.com/rails/rails/blob/7101489293ece071013417aeed2da3aa1b0927c9/activemodel/test/cases/validations/numericality_validation_test.rb#L313-L325

It looks like the most common solution in the validation tests is assert_predicate instead of using a helper method.
https://github.com/rails/rails/blob/7101489293ece071013417aeed2da3aa1b0927c9/activemodel/test/cases/validations/format_validation_test.rb#L30-L34

Maybe there's a good argument for refactoring the validation tests to match a common standard? In a separate PR? I know Rails has a [policy against cosmetic changes](https://github.com/rails/rails/pull/13771#issuecomment-32746700) but creating some form of test helper, and refactoring all the validator tests feels larger than a cosmetic change?